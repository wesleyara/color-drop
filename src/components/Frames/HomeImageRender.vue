<script setup lang="ts">
import { useFileStore } from "../../store/file";

const fileStore = useFileStore();
</script>

<template>
  <span class="relative max-h-[500px]">
    <button
      @click="fileStore.clearFile"
      class="absolute right-1 top-1 flex h-6 w-6 flex-col items-center justify-center rounded-md bg-white"
    >
      <v-icon name="md-close" />
    </button>

    <img  v-if="fileStore.file" :src="fileStore.file" alt="image uploaded" class="max-h-[500px]" />
  </span>
</template>
