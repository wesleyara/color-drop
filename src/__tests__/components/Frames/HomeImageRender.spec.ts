/* eslint-disable @typescript-eslint/no-explicit-any */
import { describe, it, expect, beforeEach } from "vitest";
import { mount, VueWrapper } from "@vue/test-utils";
import HomeImageRender from "../../../components/Frames/HomeImageRender.vue";
import { createPinia, setActivePinia } from "pinia";

describe("HomeImageRender unit tests", async () => {
  let sut: VueWrapper<any>;

  beforeEach(async () => {
    setActivePinia(createPinia());
    sut = mount(HomeImageRender);
  });

  it("should render the component", () => {
    const wrapper = sut.exists();

    expect(wrapper).toBeDefined();
  });

  it("should match the snapshot", () => {
    const wrapper = sut.html();

    expect(wrapper).toMatchSnapshot();
  });

  it("should have a icon", () => {
    const wrapper = sut.find("v-icon");

    expect(wrapper.exists()).toBe(true);
  });

  it("should have a class", () => {
    const wrapper = sut.classes();

    expect(wrapper).toContain("relative");
  });
});
