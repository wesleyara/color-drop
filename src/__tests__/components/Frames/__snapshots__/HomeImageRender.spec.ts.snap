// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeImageRender unit tests > should match the snapshot 1`] = `"<span class="relative max-h-[500px]"><button class="absolute right-1 top-1 flex h-6 w-6 flex-col items-center justify-center rounded-md bg-white"><v-icon name="md-close"></v-icon></button><!--v-if--></span>"`;
